AI/scripts/model_main_tf2.py

```diff
@@ -0,0 +1,242 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""
+# TODO Flags must be adjusted a bit.
+
+       USAGE: model_main_tf2.py [flags]
+flags:
+
+model_main_tf2.py:
+  --checkpoint_dir: Path to directory holding a checkpoint.  If `checkpoint_dir`
+    is provided, this binary operates in eval-only mode, writing resulting
+    metrics to `model_dir`.
+  --checkpoint_every_n: Integer defining how often we checkpoint.
+    (default: '50')
+    (an integer)
+  --[no]continue_from_checkpoint: whether training should continue from a
+    checkpoint
+    (default: 'false')
+  --data_dir: directory containing mosaic and CSVs to train on
+    (default: 'C:/Users/Ian/Documents/College/Fourth Year/Fall Semester/Software
+    Engineering Project/ourepository/AI/scripts\\../test/test')
+  --[no]eval_on_train_data: Enable evaluating on train data (only supported in
+    distributed training).
+    (default: 'false')
+  --eval_timeout: Number of seconds to wait for anevaluation checkpoint before
+    exiting.
+    (default: '3600')
+    (an integer)
+  --model_dir: Path to output model directory where event and checkpoint files
+    will be written.
+  --model_name: the name of the pretrained model from the TF Object Detection
+    model zoo
+    (default: 'faster_rcnn_resnet50_v1_640x640_coco17_tpu-8')
+  --name: a name that uniquely identifies this mosaic and training data
+    (default: 'test')
+  --num_train_steps: Number of train steps.
+    (an integer)
+  --num_workers: When num_workers > 1, training uses
+    MultiWorkerMirroredStrategy. When num_workers = 1 it uses MirroredStrategy.
+    (default: '1')
+    (an integer)
+  --pipeline_config_path: Path to pipeline config file.
+  --[no]record_summaries: Whether or not to record summaries during training.
+    (default: 'true')
+  --sample_1_of_n_eval_examples: Will sample one of every n eval input examples,
+    where n is provided.
+    (an integer)
+  --sample_1_of_n_eval_on_train_examples: Will sample one of every n train input
+    examples for evaluation, where n is provided. This is only used if
+    `eval_training_data` is True.
+    (default: '5')
+    (an integer)
+  --tpu_name: Name of the Cloud TPU for Cluster Resolvers.
+  --[no]use_tpu: Whether the job is executing on a TPU.
+    (default: 'false')
+"""
+
+# r"""Creates and runs TF2 object detection models.
```

Comment:
This is the old comments, so it should be removed / the stuff that is still relevant should be moved to the main comments of the file